treeherder/utils/github.py

```diff
@@ -1,17 +1,65 @@
-from treeherder.etl.common import fetch_json
+import newrelic.agent
+import requests
+from django.conf import settings
 
+from treeherder.config.settings import GITHUB_TOKEN
 
-def fetch_api(path):
-    return fetch_json("https://api.github.com/{}".format(path))
 
+def make_request(url, method='GET', headers=None, timeout=30, **kwargs):
+    """A wrapper around requests to set defaults & call raise_for_status()."""
+    headers = headers or {}
+    headers['User-Agent'] = 'treeherder/{}'.format(settings.SITE_HOSTNAME)
+    if url.find("api.github.com") > -1:
+        if GITHUB_TOKEN:
+            headers["Authorization"] = "token {}".format(GITHUB_TOKEN)
+    response = requests.request(method,
+                                url,
+                                headers=headers,
+                                timeout=timeout,
+                                **kwargs)
+    if response.history:
+        params = {
+            'url': url,
+            'redirects': len(response.history),
+            'duration': sum(r.elapsed.total_seconds() for r in response.history)
+        }
+        newrelic.agent.record_custom_event('RedirectedRequest', params=params)
 
-def compare_shas(_repo, base, head):
-    return fetch_api("repos/{}/{}/compare/{}...{}".format(_repo["owner"], _repo["repo"], base, head))
+    response.raise_for_status()
+    return response
 
 
-def commits_info(_repo):
-    return fetch_api("repos/{}/{}/commits".format(_repo["owner"], _repo["repo"]))
+def fetch_json(url, params=None):
+    response = make_request(url,
+                            params=params,
+                            headers={'Accept': 'application/json'})
+    return response.json()
 
 
-def commit_info(_repo, sha):
-    return fetch_api("repos/{}/{}/commits/{}".format(_repo["owner"], _repo["repo"], sha))
+def fetch_text(url):
```

Comment:
`make_request`, `fetch_json` and `fetch_text` are all used elsewhere so they should stay in common.py.